### PR TITLE
Adding `PartialEq` for `LogRecord` and `SpanContext` to help unit testing

### PR DESCRIPTION
**Pull request by psandana**
Fixes (NA)
Design discussion issue (if applicable) (NA)

## Changes

Added `PartialEq` derive for `LogRecord` and `SpanContext`. This way, they are comparable among their types, which is important for correct unit testing.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [X] Unit tests added/updated (if applicable)
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/logs/record.rs [hunk 1]
```diff
@@ -198,4 +199,57 @@ mod tests {
...
+    fn compare_trace_context() {
+        let span_context = SpanContext::new(
+            TraceId::from_u128(1),
+            SpanId::from_u64(1),
+            TraceFlags::default(),
+            false,
+            TraceState::default(),
+        );
+
+        let span_context_cloned = span_context.clone();
```
Comment (lalitb) [inline opentelemetry-sdk/src/logs/record.rs:213]: this test seems to be testing for `SpanContext`, and not `TraceContext`. Or I am missing something?

File: opentelemetry-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## vNext
 
+- `LogRecord` and `SpanContext` derive from `PartialEq` to facilitate Unit Testing.
```
Comment (lalitb) [inline opentelemetry-sdk/CHANGELOG.md:5]:
```suggestion
- `opentelemetry_sdk::logs::record::LogRecord` and `opentelemetry_sdk::logs::record::TraceContext` derive from `PartialEq` to facilitate Unit Testing.
```

File: opentelemetry-sdk/src/logs/record.rs (hunk 1 above)
Comment (psandana) [reply opentelemetry-sdk/src/logs/record.rs]: You are right. Just noticed copilot gave me the wrong type when creating the test skelleton and I missed to realize it.
Now should be fixed.

**Review by lalitb — APPROVED** [review]

File: opentelemetry-sdk/src/logs/record.rs [hunk 3]
```diff
@@ -198,4 +198,53 @@ mod tests {
...
+            trace_flags: Some(TraceFlags::default()),
+        };
+
+        assert_ne!(trace_context, trace_context_different);
+    }
+
+    #[test]
+    fn compare_log_record() {
+        let log_record = LogRecord {
+            event_name: Some(Cow::Borrowed("test_event")),
```
Comment (utpilla) [inline opentelemetry-sdk/src/logs/record.rs:226]: Maybe also test the equality for `Cow::Borrowed` vs `Cow::Owned`? That would ensure that we treat `Cow::Borrowed("foo")` and `Cow::Owned(String::from("foo"))` as equal and no one mistakenly overrides the `PartialEq` definition to change this behavior in future.

Comment (psandana) [reply opentelemetry-sdk/src/logs/record.rs:226]: Good point! Added.

**Review by TommyCpp — APPROVED** [review]

**Pull request merged**